python/hail/dataset.py

```diff
@@ -2679,6 +2679,61 @@ def join(self, right):
         return VariantDataset(self.hc, self._jvds.join(right._jvds))
 
     @handle_py4j
+    @record_method
+    @typecheck(datasets=tupleof(vds_type))
+    def union(*datasets):
+        """Take the union of datasets vertically (include all variants).
+
+        **Examples**
+
+        .. testsetup::
+
+            vds_autosomal = vds1
+            vds_chromX = vds1
+            vds_chromY = vds1
+
+        Union two datasets:
+
+        >>> vds_union = vds_autosomal.union(vds_chromX)
+
+        Given a list of datasets, union them all:
+
+        >>> all_vds = [vds_autosomal, vds_chromX, vds_chromY]
+
+        The following three syntaxes are equivalent:
+
+        >>> vds_union1 = vds_autosomal.union(vds_chromX, vds_chromY)
+        >>> vds_union2 = all_vds[0].union(*all_vds[1:])
+        >>> vds_union3 = VariantDataset.union(*all_vds)
+
+        **Notes**
+
+        In order to combine two datasets, four requirements must be met:
+         - the column keys must be identical (both type and identity, field order within structs matters).
+         - the row key and annotation schemas must match (field order within structs matters).
+         - the row (variant) annotation schemas must match (field order within structs matters).
```

Comment:
is the third requirement not a suset of the second one? Or am I misreading what the second requirement is?

Reply:
oops, good catch